lib/charms/blackbox_k8s/v0/blackbox_probes.py

```diff
@@ -0,0 +1,652 @@
+
+            self.blackbox_probes = BlackboxProbesProvider(
+                self,
+                probes = [
+                    ""
+                      params:
+                        module: [http_2xx]
+                      static_configs:
+                        - targets: 
+                            - http://target-a
+                            labels: 
+                                name: "device"
+                    "",
+                    ""
+                      params:
+                        module: [http_2xx_longer_timeout]
+                      static_configs:
+                        - targets: ['http://target-b' 'http://target-c']
+                    ""
+                ],
+                modules={
+                    "http_2xx_longer_timeout": {
+                        "prober": "http"
+                        "timeout": "30s"  # default is 5s
+                    }
+                },
+                refresh_event=[
+                    self.on.update_status,
+                    self.ingress.on.ready_for_unit,
+                ],
+                relation_name="blackbox-probes",
+            )
+
+        Args:
+            charm: a `CharmBase` object which manages the
+                `BlackboxProbesProvider` object. Generally, this is `self` in
+                the instantiating class.
+            probes: the probes to configure in Blackbox Exporter passed as a list
+                    of configuration probes in python structures.
+            modules: an optional definition of modules for Blackbox Exporter to
+                use. For details on how to write pass a module, see the
+                [docs upstream](https://github.com/prometheus/blackbox_exporter/blob/master/CONFIGURATION.md).
+                Further examples are provided [upstream](https://github.com/prometheus/blackbox_exporter/blob/master/example.yml).
+            refresh_event: additional `CharmEvents` event (or a list of them)
+                on which the probes and modules should be updated.
+            relation_name: name of the relation providing the Blackbox Probes
+                service. It's recommended to not change it, to ensure a
+                consistent experience across all charms that use the library.
+        """
+        self.topology = JujuTopology.from_charm(charm)
+        self._charm = charm
+        self._relation_name = relation_name
+        self._probes = [] if probes is None else probes
+        self._modules = {} if modules is None else modules
+
+        events = self._charm.on[self._relation_name]
+        self.framework.observe(events.relation_changed, self.set_probes_spec)
+
+        if not refresh_event:
+            if len(self._charm.meta.containers) == 1:
+                container = list(self._charm.meta.containers.values())[0]
+                refresh_event = [self._charm.on[container.name.replace("-", "_")].pebble_ready]
+            else:
+                refresh_event = []
```

Comment:
I would remove the unnecessary `else`:

```suggestion
            refresh_event = []
            
            if len(self._charm.meta.containers) == 1:
                container = list(self._charm.meta.containers.values())[0]
                refresh_event = [self._charm.on[container.name.replace("-", "_")].pebble_ready]
```